connectors/bitbucket-server/src/main/java/com/vmware/connectors/bitbucket/server/BitbucketServerController.java

```diff
@@ -251,7 +257,16 @@ public BitbucketServerController(WebClient rest, CardTextAccessor cardTextAccess
                         pullRequest.getProjectKey(), pullRequest.getRepositorySlug(), pullRequest.getPullRequestId())
                 .header(AUTHORIZATION, authHeader)
                 .retrieve()
-                .bodyToMono(JsonDocument.class);
+                .bodyToMono(JsonDocument.class)
+                .onErrorResume(BitbucketServerController::logUnauthorizedError);
+    }
+
+    private static <T> Mono<T> logUnauthorizedError(final Throwable throwable) {
+        if (throwable instanceof HttpClientErrorException &&
+                ((HttpClientErrorException)throwable).getStatusCode() == HttpStatus.UNAUTHORIZED) {
+            logger.error("Bitbucket server authorization failed.", throwable);
```

Comment:
This would not have solved the point of the ticket ("we aren't logging anything about the response, so it's invisible to anyone why we are getting the 401s").

I'm assuming you didn't try to test this, b/c for some reason your error handling code is never executed.

However, the new webflux changes @rworsnop put in with APF-919 do a better job printing the error response than our previous rest template code, so I'm going to close this ticket.

Example of the error output (it is the same both before and after the code in this PR):

Normal auth failure:

```
2018-03-26 10:53:08,155 ERROR [http-nio-8069-exec-2]   c.v.c.c.w.ExceptionHandlers [ExceptionHandlers.java:52] Backend returned 401 Unauthorized 
 {"errors":[{"context":null,"message":"Authentication failed. Please check your credentials and try again.","exceptionName":"com.atlassian.bitbucket.auth.IncorrectPasswordAuthenticationException"}]}
```

Auth failure due to lockout:

```
2018-03-26 10:58:46,866 ERROR [http-nio-8069-exec-8]   c.v.c.c.w.ExceptionHandlers [ExceptionHandlers.java:52] Backend returned 401 Unauthorized 
 {"errors":[{"context":null,"message":"CAPTCHA required. Your Bitbucket account has been locked. To unlock it and log in again you must solve a CAPTCHA. This is typically caused by too many attempts to login with an incorrect password. The account lock prevents your SCM client from accessing Bitbucket and its mirrors until it is solved, even if you enter your password correctly.\n\nIf you are currently logged in to Bitbucket via a browser you may need to logout and then log back in in order to solve the CAPTCHA.\n\nVisit Bitbucket at https://stash.air-watch.com for more details.","exceptionName":"com.atlassian.bitbucket.auth.CaptchaRequiredAuthenticationException"}]}
```

Reply:
I was bit hesitant to test this account lock scenario using my own credentials and I did not have test accounts.

Reply:
As the ticket mentions, you just have to log out in a browser and back in again to get prompted for the captcha.

I've locked myself out a few times now (twice when they were messing w/ vIDM settings, and once again while testing this PR) and it does no harm.